tests/integration/conftest.py

```diff
@@ -500,6 +576,86 @@ async def go_non_root_app_fixture(
     return app
 
 
+@pytest.mark.skip_juju_version("3.4")
+@pytest.fixture(scope="module", name="expressjs_app")
+def expressjs_app_fixture(
+    juju: jubilant.Juju,
+    pytestconfig: pytest.Config,
+):
+    """ExpressJS charm used for integration testing.
+    Builds the charm and deploys it and the relations it depends on.
+    """
+    app_name = "expressjs-k8s"
+
+    use_existing = pytestconfig.getoption("--use-existing", default=False)
+    if use_existing:
+        yield App(app_name)
+        return
+
+    juju.deploy(
+        "postgresql-k8s",
+        channel="14/stable",
+        base="ubuntu@22.04",
+        revision=300,
+        trust=True,
+        config={"profile": "testing"},
+    )
+    juju.wait(
+        lambda status: status.apps["postgresql-k8s"].is_active,
+        timeout=20 * 60,
+    )
+
+    resources = {
+        "app-image": pytestconfig.getoption("--expressjs-app-image"),
+    }
+    charm_file = build_charm_file(pytestconfig, "expressjs")
+    juju.deploy(
+        charm=charm_file,
+        resources=resources,
+    )
+
+    # configure postgres
+    juju.config(
+        "postgresql-k8s",
+        {
+            "plugin_hstore_enable": "true",
+            "plugin_pg_trgm_enable": "true",
+        },
+    )
+    juju.wait(lambda status: status.apps["postgresql-k8s"].is_active)
+
+    # Add required relations
+    juju.integrate(app_name, "postgresql-k8s:database")
+    juju.wait(lambda status: jubilant.all_active(status, [app_name, "postgresql-k8s"]))
+
+    yield App(app_name)
```

Comment:
If there is no cleanup that is required, i think we can just return the App right away :)
```suggestion
    return App(app_name)
```